False negatives with sourceType: module and ecmaVersion <= 5

**Tell us about your environment**

* **ESLint Version:**
v4.12.1
* **Node Version:**
8.4.0
* **npm Version:**
5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```js
module.exports = {
    "parserOptions": {
        "sourceType": "module",
        "ecmaVersion": 5,
    }
}
```

</details>

Running against this file, ecmaVersion is ignored (could also be '3'):
 
```js
// es6.js
const x = () => `hi`;
```

**What did you expect to happen?**

Either, I would see an error like the below, rejecting an unsupported config:


```bash
> eslint -c withoutModule.config.js es6.js
Incompatible config: sourceType: module cannot be used with ecmaVersion <= 5
```

or the (admittedly niche*) use-case of ES6 modules with ES5 syntax would be supported, and I'd see errors:

```bash
> eslint -c withoutModule.config.js es6.js
es6.js
  1:1  error  Parsing error: The keyword 'const' is reserved

✖ 1 problem (1 error, 0 warnings)
```

which I get correctly with this config:

```js
// withoutModule.config.js
module.exports = {
    "parserOptions": {
        "ecmaVersion": 5,
    }
}
```

*niche: I was using rollup with some old ES5 code, and only wanted to modify the code to add imports. I tried the above, and thought "cool, it's working", but actually it was silently ignoring the ecmaVersion.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors are reported:

```bash
> eslint es6.js
> echo $?
0
```

--- a/package.json
+++ b/package.json
@@ -44,7 +44,7 @@
     "eslint-scope": "^4.0.3",
     "eslint-utils": "^1.3.1",
     "eslint-visitor-keys": "^1.0.0",
-    "espree": "^5.0.1",
+    "espree": "^6.0.0-alpha.0",
     "esquery": "^1.0.1",
     "esutils": "^2.0.2",
     "file-entry-cache": "^5.0.1",

